File: src/chipschallenge/blockreactions/RedButtonTo.java
package chipschallenge.blockreactions;

import chipschallenge.Block;
import chipschallenge.Buttons;
import chipschallenge.SoundPlayer.sounds;

/**
 *
 * @author patrik
 */
public class RedButtonTo extends BlockReaction {

    private RedButtonTo() {
    }
    private static RedButtonTo mInstance = null;

    public static synchronized RedButtonTo getInstance() {
        if (mInstance == null) {
            mInstance = new RedButtonTo();
        }
        return mInstance;
    }

    public void react(Block moving, Block standing) {
        Buttons.redButtonDown(standing);
        sound().playSound(sounds.BUTTON);
    }

    public boolean canMove(Block moving, Block standing) {
        return true;
    }
}

File: ChipsChallenge.Shared/Blockreactions/RedButtonTo.cs
namespace ChipsChallenge.Shared.Blockreactions
{
    public class RedButtonTo : NoSlipReaction
    {
        private static RedButtonTo instance;

        private RedButtonTo()
        {
        }

        public static RedButtonTo Instance
        {
            get
            {
                lock (typeof(RedButtonTo))
                {
                    return instance ?? (instance = new RedButtonTo());
                }
            }
        }

        public override void React(Block moving, Block standing)
        {
            Buttons.RedButtonDown(standing);
            Sound().Play(Shared.Sound.Button);
        }

        public override bool canMove(Block moving, Block standing)
        {
            return true;
        }
    }
}